Traceback on unknown encoding

### Steps to reproduce

Run Pylint against the following code:
```python
# encoding=UTF-9
```

### Current behavior

```pytb
Traceback (most recent call last):
  File "/usr/lib/python3.7/tokenize.py", line 397, in find_cookie
    codec = lookup(encoding)
LookupError: unknown encoding: UTF-9

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/jwilk/.local/bin/pylint", line 10, in <module>
    sys.exit(run_pylint())
  File "/home/jwilk/.local/lib/python3.7/site-packages/pylint/__init__.py", line 22, in run_pylint
    PylintRun(sys.argv[1:])
  File "/home/jwilk/.local/lib/python3.7/site-packages/pylint/lint/run.py", line 349, in __init__
    linter.check(args)
  File "/home/jwilk/.local/lib/python3.7/site-packages/pylint/lint/pylinter.py", line 863, in check
    self.get_ast, self._iterate_file_descrs(files_or_modules)
  File "/home/jwilk/.local/lib/python3.7/site-packages/pylint/lint/pylinter.py", line 895, in _check_files
    for name, filepath, modname in file_descrs:
  File "/home/jwilk/.local/lib/python3.7/site-packages/pylint/lint/pylinter.py", line 952, in _iterate_file_descrs
    for descr in self._expand_files(files_or_modules):
  File "/home/jwilk/.local/lib/python3.7/site-packages/pylint/lint/pylinter.py", line 961, in _expand_files
    modules, self.config.black_list, self.config.black_list_re
  File "/home/jwilk/.local/lib/python3.7/site-packages/pylint/utils/utils.py", line 187, in expand_modules
    modparts, path=additional_search_path
  File "/home/jwilk/.local/lib/python3.7/site-packages/astroid/modutils.py", line 423, in file_info_from_modpath
    return _spec_from_modpath(modpath, path, context)
  File "/home/jwilk/.local/lib/python3.7/site-packages/astroid/modutils.py", line 648, in _spec_from_modpath
    found_spec = spec.find_spec(modpath, path)
  File "/home/jwilk/.local/lib/python3.7/site-packages/astroid/interpreter/_import/spec.py", line 337, in find_spec
    _path, modname, module_parts, processed, submodule_path or path
  File "/home/jwilk/.local/lib/python3.7/site-packages/astroid/interpreter/_import/spec.py", line 300, in _find_spec_with_path
    spec = finder.find_module(modname, module_parts, processed, submodule_path)
  File "/home/jwilk/.local/lib/python3.7/site-packages/astroid/interpreter/_import/spec.py", line 124, in find_module
    stream, mp_filename, mp_desc = imp.find_module(modname, submodule_path)
  File "/usr/lib/python3.7/imp.py", line 301, in find_module
    encoding = tokenize.detect_encoding(file.readline)[0]
  File "/usr/lib/python3.7/tokenize.py", line 426, in detect_encoding
    encoding = find_cookie(first)
  File "/usr/lib/python3.7/tokenize.py", line 405, in find_cookie
    raise SyntaxError(msg)
SyntaxError: unknown encoding for './test.py': UTF-9
```
### Expected behavior
```
test.py:1:3: E0001: unknown encoding: UTF-9 (<unknown>, line 1) (syntax-error)
```
(or something similar)

### pylint --version output
```
pylint 2.6.0
astroid 2.4.2
Python 3.7.3 (default, Jul 25 2020, 13:03:44) 
[GCC 8.3.0]
```

--- a/doc/whatsnew/fragments/3860.bugfix
+++ b/doc/whatsnew/fragments/3860.bugfix
@@ -1,0 +1,4 @@
+Fixed a syntax-error crash that was not handled properly when the declared encoding of a file
+was ``utf-9``.
+
+Closes #3860

--- a/pylint/checkers/imports.py
+++ b/pylint/checkers/imports.py
@@ -28,6 +28,7 @@ from pylint.checkers.utils import (
 )
 from pylint.exceptions import EmptyReportError
 from pylint.graph import DotBackend, get_cycles
+from pylint.interfaces import HIGH
 from pylint.reporters.ureports.nodes import Paragraph, Section, VerbatimText
 from pylint.typing import MessageDefinitionTuple
 from pylint.utils import IsortDriver
@@ -800,10 +801,10 @@ class ImportsChecker(DeprecatedMixin, BaseChecker):
                 return None
             self.add_message("relative-beyond-top-level", node=importnode)
         except astroid.AstroidSyntaxError as exc:
-            message = (
-                f"Cannot import {modname!r} due to syntax error {str(exc.error)!r}"
+            message = f"Cannot import {modname!r} due to '{exc.error}'"
+            self.add_message(
+                "syntax-error", line=importnode.lineno, args=message, confidence=HIGH
             )
-            self.add_message("syntax-error", line=importnode.lineno, args=message)
 
         except astroid.AstroidBuildingError:
             if not self.linter.is_message_enabled("import-error"):

--- a/pylint/lint/pylinter.py
+++ b/pylint/lint/pylinter.py
@@ -692,11 +692,11 @@ class PyLinter(
                     )
                     msg = get_fatal_error_message(file.filepath, template_path)
                     if isinstance(ex, AstroidError):
-                        symbol = "astroid-error"
-                        self.add_message(symbol, args=(file.filepath, msg))
+                        self.add_message(
+                            "astroid-error", args=(file.filepath, msg), confidence=HIGH
+                        )
                     else:
-                        symbol = "fatal"
-                        self.add_message(symbol, args=msg)
+                        self.add_message("fatal", args=msg, confidence=HIGH)
 
     def _check_file(
         self,
@@ -918,7 +918,8 @@ class PyLinter(
                 "syntax-error",
                 line=getattr(ex.error, "lineno", 0),
                 col_offset=getattr(ex.error, "offset", None),
-                args=str(ex.error),
+                args=f"Parsing failed: '{ex.error}'",
+                confidence=HIGH,
             )
         except astroid.AstroidBuildingError as ex:
             self.add_message("parse-error", args=ex)

--- a/tests/functional/i/import_error.txt
+++ b/tests/functional/i/import_error.txt
@@ -1,5 +1,5 @@
 import-error:3:0:3:22::Unable to import 'totally_missing':UNDEFINED
 import-error:21:4:21:26::Unable to import 'maybe_missing_2':UNDEFINED
 no-name-in-module:33:0:33:49::No name 'syntax_error' in module 'functional.s.syntax':UNDEFINED
-syntax-error:33:0:None:None::Cannot import 'functional.s.syntax.syntax_error' due to syntax error 'invalid syntax (<unknown>, line 1)':UNDEFINED
+syntax-error:33:0:None:None::Cannot import 'functional.s.syntax.syntax_error' due to 'invalid syntax (<unknown>, line 1)':HIGH
 multiple-imports:78:0:78:15::Multiple imports on one line (foo, bar):UNDEFINED

--- a/tests/functional/s/syntax/syntax_error.txt
+++ b/tests/functional/s/syntax/syntax_error.txt
@@ -1,1 +1,1 @@
-syntax-error:1:5:None:None::invalid syntax (<unknown>, line 1):UNDEFINED
+syntax-error:1:5:None:None::"Parsing failed: 'invalid syntax (<unknown>, line 1)'":HIGH

--- a/tests/test_self.py
+++ b/tests/test_self.py
@@ -569,12 +569,9 @@ class TestRunTC:
                     expected_output=expected,
                 )
 
-    def test_stdin_syntaxerror(self) -> None:
-        expected_output = (
-            "************* Module a\n"
-            "a.py:1:4: E0001: invalid syntax (<unknown>, line 1) (syntax-error)"
-        )
-
+    def test_stdin_syntax_error(self) -> None:
+        expected_output = """************* Module a
+a.py:1:4: E0001: Parsing failed: 'invalid syntax (<unknown>, line 1)' (syntax-error)"""
         with mock.patch(
             "pylint.lint.pylinter._read_stdin", return_value="for\n"
         ) as mock_stdin:

